agent/consul/internal_endpoint_test.go

```diff
@@ -784,11 +784,165 @@ func TestInternal_TerminatingGatewayServices(t *testing.T) {
 				KeyFile:     "client.key",
 			},
 		}
+
+		// Ignore raft index for equality
+		for _, s := range resp.Services {
+			s.RaftIndex = structs.RaftIndex{}
```

Comment:
Remember in unit tests this usually means you are mutating the copy of the object in memdb which can sometimes result in flaky tests if you aren't careful with the rest of the unit test.

Reply:
Though this test is fine, I just figured I'd mention it because I just had to fix a race condition related to this recently.

Reply:
Interesting, that is a good call out, hadn't thought about that before. 🤔